crates/polars-core/src/frame/mod.rs

```diff
@@ -1518,7 +1518,14 @@ impl DataFrame {
         let mut names = PlHashSet::with_capacity(cols.len());
         for name in cols {
             if !names.insert(name.as_str()) {
-                polars_bail!(duplicate = name);
+                let msg = format!(
```

Comment:
This is still to broad IMO. I think this should be here: 

https://github.com/pola-rs/polars/blob/a28b9220da58e038adeaf63d46225fa96d05f161/crates/polars-lazy/src/physical_plan/executors/projection_utils.rs#L239